### PR TITLE
Allow * expression for quarkus:add-extension

### LINKED ISSUE #1234: Allow * expression for quarkus:add-extension
Allow * expression so users can add multiple extensions in one command

For example  adding all MP/SmallRye related extensions - `mvn quarkus:add-extension -Dextensions="SmallRye*"`

### PR DESCRIPTION
**Pull request by Dufgui**
fix #1234

### CODE REVIEW

File: devtools/common/src/test/java/io/quarkus/cli/commands/AddExtensionsTest.java [hunk 1]
```diff
@@ -64,6 +64,38 @@ void testPartialMatches() throws IOException {
...
+        hasDependency(model, "quarkus-smallrye-metrics");
+        hasDependency(model, "quarkus-smallrye-reactive-messaging-kafka");
+        hasDependency(model, "quarkus-smallrye-health");
+        hasDependency(model, "quarkus-smallrye-openapi");
+        hasDependency(model, "quarkus-smallrye-jwt");
+        hasDependency(model, "quarkus-smallrye-context-propagation");
+        hasDependency(model, "quarkus-smallrye-reactive-type-converters");
+        hasDependency(model, "quarkus-smallrye-reactive-messaging-amqp");
+        hasDependency(model, "quarkus-smallrye-fault-tolerance");
+    }
```
Comment (cescoffier) [inline devtools/common/src/test/java/io/quarkus/cli/commands/AddExtensionsTest.java]: We need to test with more complicated regexps as all the escapes are uncovered.
Also, we need to check with non matching regexp

**Review by cescoffier — CHANGES_REQUESTED** [review]
I've made a comment about the tests.
This should also be documented somewhere.

File: devtools/common/src/test/java/io/quarkus/cli/commands/AddExtensionsTest.java (hunk 1 above)
Comment (cescoffier) [reply devtools/common/src/test/java/io/quarkus/cli/commands/AddExtensionsTest.java]: What does happen in the case of `**`?

**Comment by Dufgui** [comment]
I add doc, and I fix the conf about the plugin in gradle and missing doc of addExtension
@cescoffier @aloubyansky

**Review by cescoffier — APPROVED** [review]

**Pull request merged**